go-controller/pkg/ovn/service.go

```diff
@@ -165,6 +167,130 @@ func (ovn *Controller) syncServices(services []interface{}) {
 	}
 }
 
+func (ovn *Controller) createService(service *kapi.Service) error {
+	klog.V(5).Infof("Creating service %s", service.Name)
+	if !util.IsClusterIPSet(service) {
+		return fmt.Errorf("no cluster IP for service %s found", service.Name)
+	} else if len(service.Spec.Ports) == 0 {
+		return fmt.Errorf("no Ports specified for service")
+	}
+
+	for _, svcPort := range service.Spec.Ports {
+		var port int32
+		if util.ServiceTypeHasNodePort(service) {
+			port = svcPort.NodePort
+		} else {
+			port = svcPort.Port
+		}
+		if port == 0 {
+			continue
+		}
+
+		protocol := svcPort.Protocol
+		if protocol == "" || (protocol != TCP && protocol != UDP) {
+			protocol = TCP
+		}
+
+		if util.ServiceTypeHasNodePort(service) {
+			// Each gateway has a separate load-balancer for N/S traffic
+
+			physicalGateways, _, err := ovn.getOvnGateways()
+			if err != nil {
+				return err
+			}
+
+			for _, physicalGateway := range physicalGateways {
+				loadBalancer, err := ovn.getGatewayLoadBalancer(physicalGateway,
```

Comment:
@trozet don't wrap this line if you've got longer ones below :)

Reply:
I think when i was switching around which getgatewayloadbalancer function I wanted to use (util vs ovn) I ended up shortening this. Will fix the wrap.